app/models/katello/activation_key.rb

```diff
@@ -98,7 +98,11 @@ def products
       if cp_pools
         pools = cp_pools.collect { |cp_pool| Pool.find_by(:cp_id => cp_pool['id']) }
         pools.each do |pool|
-          all_products << pool.subscription.products
+          if pool.subscription
+            all_products << pool.subscription.products
+          else
+            Rails.logger.error("Pool #{pool.id} is missing its subscription id.")
```

Comment:
I'm :+1: for logging and moving on in the loop, since the error is not fatal. IMO this behavior lowers the severity of the issue, and users can reach out for help if they notice anomalies and would like assistance.